tests/src/python/provider_python.py

```diff
@@ -387,7 +387,7 @@ def supportsSubsetString(self):
     def createSpatialIndex(self):
         if self._spatialindex is None:
             self._spatialindex = QgsSpatialIndex()
-            for f in self._features:
+            for f in self._features.values():
```

Comment:
Iterating over keys meant we were passing an id into `insertFeature()` which accepts either a feature OR a (feature, bounds)